File: src/main/java/app/askresume/api/job/facade/JobFacade.java
package app.askresume.api.job.facade;

import app.askresume.api.job.dto.response.JobResponse;
import app.askresume.domain.job.service.JobService;
import app.askresume.domain.locale.constant.LocaleType;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class JobFacade {

    private final JobService jobService;

    public List<JobResponse> findJobs(LocaleType locale) {
        return jobService.findJobs(locale);
    }
}

File: src/main/kotlin/app/askresume/api/job/facade/JobFacade.kt
package app.askresume.api.job.facade

import app.askresume.api.job.mapper.JobMapper
import app.askresume.api.job.vo.JobResponse
import app.askresume.domain.job.service.JobReadOnlyService
import app.askresume.domain.locale.constant.LocaleType
import app.askresume.global.annotation.Facade

@Facade
class JobFacade(
    private val jobReadOnlyService: JobReadOnlyService,
    private val jobMapper: JobMapper,
) {

    fun findJobs(locale: LocaleType): List<JobResponse> {
        return jobMapper.jobResponseListOf(jobReadOnlyService.findJobs(locale))
    }
}